frontend/src/metabase/visualizations/components/ChartSettings/ChartSettings.tsx

```diff
@@ -0,0 +1,409 @@
+    // NOTE: special props to support adding additional fields
+    question,
+    onShowWidget: handleShowWidget,
+    onEndShowWidget: handleEndShowWidget,
+    currentSectionHasColumnSettings,
+    columnHasSettings,
+    onChangeSeriesColor: handleChangeSeriesColor,
+  };
+
+  const onResetToDefault =
+    // resetting virtual cards wipes the text and broke the UI (metabase#14644)
+    !_.isEqual(chartSettings, {}) && (chartSettings || {}).virtual_card == null
+      ? handleResetSettings
+      : null;
+
+  const showSectionPicker =
+    // don't show section tabs for a single section
+    sectionNames.length > 1 &&
+    // hide the section picker if the only widget is column_settings
+    !(
+      visibleWidgets.length === 1 &&
+      visibleWidgets[0].id === "column_settings" &&
+      // and this section doesn't have that as a direct child
+      !currentSectionHasColumnSettings
+    );
+
+  return (
+    <ChartSettingsRoot className={className}>
+      <ChartSettingsMenu data-testid="chartsettings-sidebar">
+        {showSectionPicker && (
+          <SectionContainer isDashboard={false}>
+            <Radio
+              value={chartSettingCurrentSection ?? undefined}
+              onChange={handleShowSection}
+              options={sectionNames}
+              optionNameFn={v => v}
+              optionValueFn={v => v}
+              optionKeyFn={v => v}
+              variant="underlined"
+            />
+          </SectionContainer>
+        )}
+        <ChartSettingsListContainer className={CS.scrollShow}>
+          <ChartSettingsWidgetList
+            widgets={visibleWidgets}
+            extraWidgetProps={extraWidgetProps}
+          />
+        </ChartSettingsListContainer>
+      </ChartSettingsMenu>
+      {!noPreview && (
+        <ChartSettingsPreview>
+          <SectionWarnings warnings={warnings} size={20} />
+          <ChartSettingsVisualizationContainer>
+            <Visualization
+              className={CS.spread}
+              rawSeries={chartSettingsRawSeries}
+              showTitle
+              isEditing
+              isDashboard
+              dashboard={dashboard}
+              dashcard={dashcard}
+              isSettings
+              showWarnings
+              onUpdateVisualizationSettings={handleChangeSettings}
+              onUpdateWarnings={setWarnings}
+            />
+          </ChartSettingsVisualizationContainer>
+          <ChartSettingsFooter
+            onDone={handleDone}
+            onCancel={handleCancel}
+            onReset={onResetToDefault}
+          />
+        </ChartSettingsPreview>
+      )}
+      <ChartSettingsWidgetPopover
+        anchor={popoverRef as HTMLElement}
+        widgets={[styleWidget, formattingWidget].filter(
+          (widget): widget is Widget => !!widget,
+        )}
+        handleEndShowWidget={handleEndShowWidget}
+      />
+    </ChartSettingsRoot>
+  );
+};
+
+export const ChartSettingsWithState = (props: ChartSettingsWithStateProps) => {
```

Comment:
This does not fully translate the behavior of `withTransientSettingState` because the hoc updated the component state when the prop changed

Reply:
cool, I'll add a `useEffect` here that updates when props.settings changes.